sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java

```diff
@@ -0,0 +1,106 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.app.Application;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.view.View;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.sentry.EventProcessor;
+import io.sentry.SentryEvent;
+import io.sentry.android.core.internal.util.MainThreadChecker;
+import io.sentry.util.Objects;
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.IOException;
+// import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public final class ScreenshotEventProcessor
+    implements EventProcessor, Application.ActivityLifecycleCallbacks, Closeable {
+
+  private final @NotNull Application application;
+  private final @NotNull SentryAndroidOptions options;
+  //  TODO: private final @NotNull WeakReference<Activity> currentActivity = new WeakReference();
+  private @Nullable Activity activity;
```

Comment:
Multiple screens app would be a problem, but we don't need to tackle that now, Use `WeakReference` for not leaking it.

Reply:
Do you mean multi-window?

Reply:
There are a few gotchas https://developer.android.com/guide/topics/large-screens/multi-window-support
Depending on the OS version, `currentActivity` might be Activity A or B.